provider/consul/consul_catalog.go

```diff
@@ -397,6 +393,26 @@ func (p *CatalogProvider) getBasicAuth(tags []string) []string {
 	return []string{}
 }
 
+func (p *CatalogProvider) hasStickinessLabel(tags []string) bool {
+	stickiness := p.getTag(types.LabelBackendLoadbalancerStickiness, tags, "")
+
+	sticky := p.getTag(types.LabelBackendLoadbalancerSticky, tags, "")
+	if len(sticky) != 0 {
+		log.Warn("Deprecated configuration found: %s. Please use %s.", types.LabelBackendLoadbalancerSticky, types.LabelBackendLoadbalancerStickiness)
+	}
+
+	return (len(stickiness) != 0 && strings.EqualFold(strings.TrimSpace(stickiness), "true")) ||
+		(len(sticky) != 0 && strings.EqualFold(strings.TrimSpace(sticky), "true"))
```

Comment:
`main(int c,char**v){return!m(v[1],v[2]);}m(char*s,char*t){return*t-42?*s?63==*t|*s==*t&&m(s+1,t+1):!*t:m(s,t+1)||*s&&m(s+1,t);}`

Also, `len(stickiness) > 0` & `len(sticky) > 0` would be better :)

Reply:
2 symmetric lines (on 2 lines) separate by a `||`, seems reasonable to me.
But I will change.

Reply:
Even if it's symmetric, it's hard/slow to read.